### DISCUSSION TITLE
LDAP auth: `Network error: Unexpected token '<', "<html> <h"... is not valid JSON`

### DISCUSSION THREAD

**Discussion by jko-nic** [requarks/wiki#6688] [Error / Bug Report] [answered]
Hi,
when trying LDAP auth, I encountered error

`Network error: Unexpected token '<', "<html> <h"... is not valid JSON`

which appears after a while when trying to log in. I am sure my LDAP setting are fully working, but I cannot prove it, because neither setting `logLevel: silly` nor setting LDAP Debug Flag do not produce any additional output.

In my opinion, I think wiki.js awaits LDAP response as a JSON, but maybe gets HTML (?) instead, and this is confusing. Any other web application is working with our ldap, so can you please help me inspect and, in ideal case, solve this issue I am facing?

There are no other errors, only warning with every attempt to login saying `warn: input.replace is not a function`. It is clear installation from yesterday, we are testing wiki.js to be our main wiki system.

Thanks in advance,
jko-nic

---
_Additional info:
OS: Debian 12, fully updated, running as VM in Proxmox
Wiki.js: 2.5.300 (latest stable AFAIK), installed using [linux method](https://docs.requarks.io/install/linux)
Postgresql: 15.3 on localhost
Node.js: 18.13.0+dfsg1-1
LDAP: OpenLDAP, version 2.4.44+dfsg-5+deb9u4, running on debian 9 (if it matters)_

**Comment by NGPixel** [comment]
Are you using a reverse proxy (e.g. Cloudflare, nginx, etc.?) Something is blocking the request.

**Reply by jko-nic** [reply]
Yes, using nginx. Maybe I messed up `proxy_set_header` options (or other security headers), I will investigate it later. Thanks for hint, I will certainly reply back.

**Reply by jko-nic** [reply]
I think I repaired headers, but to troubleshoot it more deeply I stopped and disabled nginx and set up (in `config.yml`) SSL like this (both cert and key are owned and fully accessible by wikijs user):

![obrazek](https://github.com/requarks/wiki/assets/74405524/53bb7601-b2f9-44a1-8095-2ff8f7a1ce96)

HTTPS works, but the same error (when using LDAP auth) is still present (Local auth works OK).

**Comment by wugren** [comment]
I also encountered the same problem. It will appear after the new installation. It is an error when the jwt signature is generated after the server login is completed.
error: secretOrPrivateKey is not valid key material
code location：
  return {
    token: jwt.sign({
      id: user.id,
      email: user.email,
      name: user.name,
      av: user.pictureUrl,
      tz: user.timezone,
      lc: user.localeCode,
      df: user.dateFormat,
      ap: user.appearance,
      // defaultEditor: user.defaultEditor,
      permissions: user.getGlobalPermissions(),
      groups: user.getGroups()
    }, {
      key: WIKI.config.certs.private,
      passphrase: WIKI.config.sessionSecret
    }, {
      algorithm: 'RS256',
      expiresIn: WIKI.config.auth.tokenExpiration,
      audience: WIKI.config.auth.audience,
      issuer: 'urn:wiki.js'
    }),
    user
  }

**Reply by jko-nic** [reply]
Thanks for reply. Did you achieve to fix it?

**Reply by wugren** [reply]
I encountered the issue when installing on QNAP NAS. Later, I found out that using node18 to call crypto.generateKeyPairSync to generate a key and then calling crypto.createPrivateKey would result in the error. It should be that node18 is somewhat incompatible with QNAP's system, and now I have switched to a different machine without any issues

**Comment by jko-nic** [comment, marked as answer]
The error was probably in my Group settings - when I disabled it (as my coworker suggested), it suddenly worked.

So forget me, please, my incompetence and thank you for your time spent on this. I will continue to debug our LDAP groups.

Thanks again for your work and support and have a nice day without users like me! :-)
